source/utils/types.ts

```diff
@@ -44,43 +49,54 @@ export type StatusCode =
 	| 503
 	| 504;
 
-export type NextFunction = (error?: Error | string) => void;
+export type DeepPartial<T> = {
```

Comment:
Since you're using this type, maybe you could comment on https://github.com/sindresorhus/type-fest/issues/18 with some use-cases you've found for it. 🙏

Reply:
I personally can't think of many use cases for the type, except for very complex types that can have everything fully optional when calling a function, and that nests deep.